src/onemod/pipeline.py

```diff
@@ -19,26 +19,30 @@
 
 
 class Pipeline(BaseModel):
-    """
-    Notes
-    -----
-    * `data` is raw input data used to figure out grouped stage subsets
-       when initializing the pipeline; stages will get their data from
-       other stages (with the exception of any preprocessing stages)
-    * `data` is only required if using grouped stages
-    * TODO: Define dependencies between stages
-    * TODO: DAG creation, validation
-    * TODO: Run locally or on Jobmon
+    """Pipeline class.
+
+    Attributes
+    ----------
+    name : str
+        Pipeline name.
+    config : PipelineConfig
+        Pipeline configuration.
+    directory : Path
+        Experiment directory.
+    data : Path or None, optional
+        Input data used to create data subsets. Required for pipeline or
+        stage `groupby` attribute. Default is None.
+    groupby : set[str] or None, optional
+        ID names used to create data subsets. Default is None.
 
     """
 
     name: str
     config: PipelineConfig
-    directory: Path
+    directory: Path  # TODO: replace with DataInterface
```

Comment:
Were you thinking of using `DataInterface` from `pplkit` as an imported module or we should create a module within OneMod that takes inspiration from it?

Reply:
I was thinking of importing it from pplkit